common/cache/src/main/java/org/thingsboard/server/cache/CaffeineCacheConfiguration.java

```diff
@@ -72,13 +95,41 @@ public CacheManager cacheManager() {
         return new TransactionAwareCacheManagerProxy(manager);
     }
 
+    @PostConstruct
+    public void init() {
+        if (cacheStatsEnabled) {
+            log.debug("initializing cache stats scheduled job");
+            scheduler = Executors.newSingleThreadScheduledExecutor(ThingsBoardThreadFactory.forName("cache-stats"));
```

Comment:
We need to extract the ScheduledExecutor to a component and use this component for all sorts of periodic logging/stats tasks. This will save many threads.